query/functor.hpp

```diff
@@ -575,6 +579,86 @@ struct UnaryFunctor<UUIDT, UUIDT> {
   }
 };
 
+// Specialize UnaryFunctor for GeoPointT input type to types other than
+// GeoPointT
+template <typename O>
+struct UnaryFunctor<
+    O,
+    GeoPointT,
+    typename std::enable_if<
+        !std::is_same<O, GeoPointT>::value && !std::is_same<O, UUIDT>::value
+    >::type
+  >{
+  typedef thrust::tuple<GeoPointT, bool> argument_type;
+  typedef thrust::tuple<O, bool> result_type;
+
+  explicit UnaryFunctor(UnaryFunctorType functorType)
+      : functorType(functorType) {}
+
+  UnaryFunctorType functorType;
+
+  __host__ __device__ result_type operator()(const argument_type t) const {
+       O o;
```

Comment:
Maybe a cleaner way to do this is disable binding non-geopoint output type in dimension_transform.cu

Reply:
sorry I didn't get it. we do still want to support other types as dimensions.